bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/handler/ItemStateTriggerHandler.java

```diff
@@ -42,6 +42,7 @@
 
     private String itemName;
     private String state;
+    private String oldState;
```

Comment:
The class ItemStateTriggerHandler currently handles changed events as well as normal events, so I think this is correct. But you are right, I have added it at the ItemStateEvent as well, that is wrong.